src/main/java/com/ericsson/ei/mongo/MongoDBHandler.java

```diff
@@ -261,19 +263,14 @@ public boolean isMongoDBServerUp() {
 
     }
 
-    // Establishing the connection to mongodb and creating a collection
-    private void createConnection() {
-        if (!StringUtils.isBlank(mongoProperties.getUsername())
-                && !StringUtils.isBlank(new String(mongoProperties.getPassword()))) {
-            ServerAddress address = new ServerAddress(mongoProperties.getHost(),
-                    mongoProperties.getPort());
-            MongoCredential credential = MongoCredential.createCredential(
-                    mongoProperties.getUsername(),
-                    mongoProperties.getDatabase(), mongoProperties.getPassword());
-            mongoClient = new MongoClient(address, Collections.singletonList(credential));
-        } else {
-            mongoClient = new MongoClient(mongoProperties.getHost(), mongoProperties.getPort());
+    private void createMongoClient() throws AbortExecutionException {
```

Comment:
Is this right or is it supposed to be throwing MongoConfigurationException?

Reply:
It is a RuntimeException.